src/misc.js

```diff
@@ -82,9 +83,46 @@ export function hashIsEqual(h1, h2) {
 }
 
 export function cloneHasher(h) {
-    const ph = h.getPartialHash();
-    const res = Blake2b(64);
-    res.setPartialHash(ph);
+    return h.clone();
+}
+
+export function fromPartialHash(partial) {
+    // NOTE: this is unsafe and uses internal API
```

Comment:
Hello @paulmillr!
Thank you for your contribution! I imagine it was a lot of work!
Can you please elaborate on this comment? Why is it unsafe? 
Just because of the usage of internal API or there are other considerations?

Reply:
It’s because it uses internal api.

stuff like this is never exposed in libraries.

Reply:
Why do you need "partial hashes"?

Reply:
I don't know. I will ask Jordi.

Reply:
Jordi's response:
> We are using the zCash protocol, and are constrained as we want to be compatible to the protocol without changing it. With partial hashes, it allows us to verify the presence of a specific contribution without having to download all the contributions.